applications/fishing-map/src/features/analysis/Analysis.worker.ts

```diff
@@ -0,0 +1,39 @@
+import type { Geometry, Polygon, MultiPolygon } from 'geojson'
+import booleanContains from '@turf/boolean-contains'
+import booleanOverlap from '@turf/boolean-overlap'
+import simplify from '@turf/simplify'
+
+type Feature = GeoJSON.Feature<GeoJSON.Geometry>
+export type FilteredPolygons = {
+  contained: Feature[]
+  overlapping: Feature[]
+}
+
+export function filterByPolygon(features: Feature[], polygon: Geometry): FilteredPolygons {
+  const simplifiedPoly = simplify(polygon as Polygon | MultiPolygon, { tolerance: 0.1 })
+  const filtered = features.reduce(
+    (acc, feature) => {
+      const isContained =
+        simplifiedPoly.type === 'MultiPolygon'
+          ? simplifiedPoly.coordinates.some((coordinates) =>
+              booleanContains({ type: 'Polygon', coordinates }, feature.geometry as Polygon)
+            )
+          : booleanContains(simplifiedPoly, feature.geometry as Polygon)
+
+      if (isContained) {
+        acc.contained.push(feature)
+      } else {
+        // TODO try to get the % of overlapping to use it in the value calculation
```

Comment:
I doubt getting the % of overlapping and use that in the value calculation would be a better approximation, it could be totally possible that the overlapping area does not match the area of the box where apparent fishing activity is happening since fishing activity in the cell could be not evenly distributed (I think this is the most common)

Reply:
Indeed, this needs further investigation... we'll need to brainstorm on this

Reply:
I agree with @enriquetuya, we might evaluate getting the chart data from backend. I think it should provide better accuracy and performance.